### PR TITLE
Added pflag support in cli

### LINKED ISSUE #1253: [Housekeeping] Flyte copilot should not need explicit configuration - flyte-data-config
**Describe the Issue**
Related to #1251.
Flyte co-pilot currently needs an explicit configmap to control the storage configuration. this is only required, because till recently flytestdlib did not support commandline updates to config vars. This has now been fixed.

This issue proposes that we upgrade copilot to new config and then update flyteplugins to pass the default flytepropeller storage configuration, with few modifications as command line parameters.

Controlled from here https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/flytek8s/copilot.go. Delete this https://github.com/flyteorg/flyte/blob/master/helm/templates/_helpers.tpl#L207-L216

Modifications:
 enable-multicontainer: True
 default download limit.

We should also adjust the StorageSize based on user provided information. 
https://github.com/flyteorg/flyteplugins/blob/master/go/tasks/pluginmachinery/flytek8s/copilot.go#L214 and default to a reasonable value

**What if we do not do this?**
CoPilot cannot download a lot of data.

***Related component***
FlytePlugins. Flyte-copilot

### PR DESCRIPTION
**Pull request by yindia**
# TL;DR
Added pflag support for a dependent pr https://github.com/flyteorg/flyteplugins/pull/191/files

- bump new version of flytestdlib for flag support

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
 _How did you fix the bug, make the feature etc. Link to any design docs etc_

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/1253

## Follow-up issue
_NA_
OR
_https://github.com/flyteorg/flyte/issues/<number>_

### CODE REVIEW

File: go.sum [hunk 1]
```diff
@@ -192,6 +192,10 @@ github.com/flyteorg/flytestdlib v0.3.13 h1:5ioA/q3ixlyqkFh5kDaHgmPyTP/AHtqq1K/TI
 github.com/flyteorg/flytestdlib v0.3.13/go.mod h1:Tz8JCECAbX6VWGwFT6cmEQ+RJpZ/6L9pswu3fzWs220=
 github.com/flyteorg/flytestdlib v0.3.20 h1:1BTg02U1dyVZduxuViebHkTrw+4Owc5TI/lPbHiimzE=
 github.com/flyteorg/flytestdlib v0.3.20/go.mod h1:VlbQuHTE+z2N5qusfwi+6WEkeJoqr8Q0E4NtBAsdwkU=
+github.com/flyteorg/flytestdlib v0.3.29 h1:Ml2Ed9hwHWcb5EmBIebu8SUWDNMJn8NOxjeemNPDqAY=
```
Comment (pmahindrakar-oss) [inline go.sum]: Can we run go mod tidy on this change

File: cmd/root.go [hunk 2]
```diff
@@ -162,17 +162,19 @@ func NewDataCommand() *cobra.Command {
...
 
-func (r *RootOptions) initConfig(_ *cobra.Command, _ []string) error {
+func (r *RootOptions) initConfig(cmd *cobra.Command, _ []string) error {
 	r.configAccessor = viper.NewAccessor(config.Options{
 		StrictMode:  true,
 		SearchPaths: []string{r.cfgFile},
 	})
 
+	// persistent flags were initially bound to the root command so we must bind to the same command to avoid
+	r.configAccessor.InitializePflags(cmd.PersistentFlags())
```
Comment (pmahindrakar-oss) [inline cmd/root.go:172]: Are there no test changes required for this change

**Review by pmahindrakar-oss — APPROVED** [review]

File: cmd/root.go (hunk 2 above)
Comment (yindia) [reply cmd/root.go:172]: no we don't have test fro root

Comment (pmahindrakar-oss) [reply cmd/root.go:172]: Is it too difficult to add one for this function. If yes then we can defer it or else would be good to improve on this aspect
